tests/test_xss_exposure.py

```diff
@@ -0,0 +1,47 @@
+"""
+Tests for identifying XSS vulnerabilities.
+This is currently done with the "q" method is called.
+"""
+
+import os
+from mock import patch
+
+from bok_choy.web_app_test import WebAppTest
+from .pages import SitePage
+from bok_choy.page_object import XSSExposureError
+
+
+class XSSExposureTest(WebAppTest):
+    """
+    Tests for identifying XSS vulnerabilities.
+    """
+    def _visit_page(self, page_name):
+        self.site_page = SitePage(self.browser)
+        self.site_page.name = page_name
+        self.site_page.visit()
+
+    @patch.dict(os.environ, {'VERIFY_XSS': 'True'})
+    def test_html_exposure(self):
+        self._visit_page("xss_html")
+        with self.assertRaisesRegexp(XSSExposureError, "2 XSS issue"):
+            self.site_page.q(css='.unescaped')
+
+    @patch.dict(os.environ, {'VERIFY_XSS': 'True'})
+    def test_js_exposure(self):
+        self._visit_page("xss_js")
+        with self.assertRaisesRegexp(XSSExposureError, "1 XSS issue"):
+            self.site_page.q(css='.unescaped')
+
+    @patch.dict(os.environ, {'VERIFY_XSS': 'True'})
+    def test_escaped(self):
```

Comment:
It would make sense to also add a testcase for a page that has mixed safe and unsafe areas.